wgpu/src/api/shader_module.rs

```diff
@@ -236,3 +236,21 @@ pub struct ShaderModuleDescriptorSpirV<'a> {
     pub source: Cow<'a, [u32]>,
 }
 static_assertions::assert_impl_all!(ShaderModuleDescriptorSpirV<'_>: Send, Sync);
+
+/// Descriptor for a shader module given by Metal MSL sourc, for use with
+/// [`Device::create_shader_module_msl`].
+///
+/// This type is unique to the Rust API of `wgpu`. In the WebGPU specification,
+/// only WGSL source code strings are accepted.
```

Comment:
It would be great to get some description of the shader ABI we present to shaders and how things get translated. It doesn't need to be as strict as in wgpu-hal, but should guide people on what wgpu expects the shader to have.